tests/api/cyto_dl_model/test_segmentation_plugin_model.py

```diff
@@ -0,0 +1,134 @@
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+import pytest
+
+from cyto_dl.api.cyto_dl_model import SegmentationPluginModel
+
+
+# Taken together, these tests verify that the SegmentationPluginModel aligns with the
+# default segmentation_plugin config file. It does this by a) verifying that methods which
+# attempt to change config items on a bad (fake) config throw exceptions because those config
+# items do not exist and b) verifying that those methods do not throw exceptions when called on
+# a model that uses the default config file
+#
+# Note that these tests do not test whether or not the config variables set by SegmentationPluginModel
+# are actually used how we expect by Cyto-DL, just that they exist in the default config.
+@pytest.fixture
+def model_with_default_config() -> SegmentationPluginModel:
+    return SegmentationPluginModel()
+
+
+@pytest.fixture
+def model_with_bad_config() -> SegmentationPluginModel:
+    return SegmentationPluginModel(config_filepath=Path(__file__).parent / "bad_config.yaml")
+
+
+class TestDefaultConfig:
+    @patch("cyto_dl.api.cyto_dl_model.cyto_dl_base_model.train_model")
+    def test_train_no_ckpt(
+        self, train_model_mock: Mock, model_with_default_config: SegmentationPluginModel
+    ):
+        model_with_default_config.train(1, "manifest", "output_dir")
+        train_model_mock.assert_called_once()
+
+    @patch("cyto_dl.api.cyto_dl_model.cyto_dl_base_model.train_model")
+    def test_train_with_ckpt(
+        self, train_model_mock: Mock, model_with_default_config: SegmentationPluginModel
+    ):
+        model_with_default_config.train(1, "manifest", "output_dir", checkpoint=Path("ckpt"))
+        train_model_mock.assert_called_once()
```

Comment:
It seems to me like the assert for this test should be different from the one above, since the checkpoint was used.

Reply:
discussed in person.  its just a tough thing to design.